### DISCUSSION TITLE
How to handle yaml grammars

### DISCUSSION THREAD

**Discussion by serkonda7** [github-linguist/linguist#6437] [Q&A]
The grammar in https://github.com/serkonda7/vscode-vba is written in yaml (because it's just better).

It's converted to JSON before extension releases to VS Code. How to do this for linguist?

**Comment by lildude** [comment]
> It's converted to JSON before extension releases to VS Code. How to do this for linguist?

We expect the grammars to be in the format and locations ready to use by the respective editors, without processing, as seen in the code here:

https://github.com/github-linguist/linguist/blob/8a2415b96ad796af16e853ddb091521af919c486/tools/grammars/compiler/loader.go#L119-L128

Anything else is ignored. If you were to try to add that grammar to linguist, the script will report an error because the grammar can't be used as-is directly from the repo.

We then convert these to JSON for our highlighting engine, if not already in JSON format.

**Comment by serkonda7** [comment]
Thanks for the answer.

I am trying around with renaming the grammar, so it's recognized by linguist and ran into another question/problem:

`script/add-grammar https://github.com/serkonda7/vscode-vba` does not pull the most recent commit to the vscode-vba repo. Is there some caching or so?

**Reply by lildude** [reply]
> Is there some caching or so?

Nope. The script doesn't do any caching itself. All it does is add it as a submodule and will use the commit at the time of first invocation. If you want to use the latest commit, you need to update the submodule as you would normally for git submodules.

**Reply by serkonda7** [reply]
Thanks
